Java/PlatinumGenomes-variant-transformation/src/main/java/com/google/cloud/genomics/examples/TransformNonVariantSegmentData.java

```diff
@@ -131,34 +160,148 @@ public void processElement(ProcessContext c) {
-          .set("end", v.getEnd())
-          .set("reference_bases", v.getReferenceBases())
-          .set("alternate_bases", v.getAlternateBases())
-          .set("names", (v.getNames() == null) ? new ArrayList<String>() : v.getNames())
-          .set("filter", (v.getFilter() == null) ? new ArrayList<String>() : v.getFilter())
-          .set("quality", v.getQuality())
-          .set("call", calls);
+      TableRow row =
+          new TableRow()
+              .set("variant_id", v.getId())
+              .set("reference_name", v.getReferenceName())
+              .set("start", v.getStart())
+              .set("end", v.getEnd())
+              .set("reference_bases", v.getReferenceBases())
+              .set("alternate_bases",
+                  (v.getAlternateBases() == null) ? new ArrayList<String>() : v.getAlternateBases())
+              .set("names", (v.getNames() == null) ? new ArrayList<String>() : v.getNames())
+              .set("filter", (v.getFilter() == null) ? new ArrayList<String>() : v.getFilter())
+              .set("quality", v.getQuality()).set("call", calls)
+              .set(HAS_AMBIGUOUS_CALLS_FIELD, v.getInfo().get(HAS_AMBIGUOUS_CALLS_FIELD).get(0));
+
       c.output(row);
     }
   }
 
+  /**
+   * Pipeline function to filter out calls for specific callSetNames.
+   * 
+   * One may want to do this for a list of callsets that have failed quality control checks.
+   */
+  public static final class FilterCallsFn extends DoFn<Variant, Variant> {
+    
+    private final ImmutableSet<String> callSetNamesToExclude;
+    
+    /**
+     * @param callSetNamesToExclude
+     */
+    public FilterCallsFn(ImmutableSet<String> callSetNamesToExclude) {
+      super();
+      this.callSetNamesToExclude = callSetNamesToExclude;
+    }
+
+    @Override
+    public void processElement(ProcessContext context) {
+      Variant variant = context.element();
+
+      // We may have variants without calls if any callsets were deleted from the variant set.
+      if(!(null == variant.getCalls() || variant.getCalls().isEmpty())) {
+        List<Call> filteredCalls =
+            Lists.newArrayList(Iterables.filter(variant.getCalls(), new Predicate<Call>() {
+              @Override
+              public boolean apply(Call call) {
+                if (callSetNamesToExclude.contains(call.getCallSetName())) {
+                  return false;
+                }
+                return true;
+              }
+            }));
+        variant.setCalls(filteredCalls);
+      }
+
+      context.output(variant);
+    }
+  }
+  
+  /**
+   * Pipeline function to flag any variants with more than one call for a particular callSetName.
+   * 
+   * We don't see this for the tidy test datasets such as Platinum Genomes. But in practice, we have
+   * seen datasets with the same individual sequenced twice, mistakes in data conversions prior to
+   * this step, etc... So this function emits and aggregate counter of the number of variants with
```

Comment:
typo: should "an aggregate counter" instead of "and". 
But perhaps we should rephrase. "emits" suggests that this is an output of this pipeline stage. But what's going here instead is that we're using Dataflow's special "Aggregator" class which not only aggregates but also only shows its value in the UI. This value is not an output and (AFAIK) can't be read programatically. When used in lowercase one might think we're using the ordinary meaning of aggregation.

How about: "So this function flags the particular variants with ambiguous calls for the same individual, and also updates the UI with a count of such variants."